File: Set Matrix Zeroes/Set Matrix Zeroes.java
public class Solution {
    public void setZeroes(int[][] matrix) {
        if(matrix.length == 0 || matrix[0].length == 0)     return;
        
        boolean[] row = new boolean[matrix.length];
        boolean[] col = new boolean[matrix[0].length];
        
        for(int i = 0; i < matrix.length; ++i)              row[i] = false;
        for(int i = 0; i < matrix[0].length; ++i)           col[i] = false;
        
        for(int i = 0; i < matrix.length; ++i){
            for(int j = 0; j < matrix[0].length; ++j){
                if(matrix[i][j] == 0){
                    row[i] = true;
                    col[j] = true;
                }
            }
        }
        
        for(int i = 0; i < matrix.length; ++i){
            if(row[i]){
                for(int j = 0; j < matrix[0].length; ++j){
                    matrix[i][j] = 0;
                }
            }
        }
        for(int j = 0; j < matrix[0].length; ++j){
            if(col[j]){
                for(int i = 0; i < matrix.length; ++i){
                    matrix[i][j] = 0;
                }
            }
        }
    }
}

File: Set Matrix Zeroes/Set Matrix Zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int> > &matrix) {
        std::set<int> zeroRows;
        std::set<int> zeroCols;
        for(unsigned int i = 0; i < matrix.size(); ++i){
            for(unsigned int j = 0; j < matrix[i].size(); ++j){
                if(!matrix[i][j]){
                    zeroRows.insert(i);
                    zeroCols.insert(j);
                }
            }
        }
        
        for(unsigned int i = 0; i < matrix.size(); ++i){
            for(unsigned int j = 0; j < matrix[i].size(); ++j){
                if(zeroRows.find(i) != zeroRows.end()){
                    matrix[i][j] = 0;
                }
                if(zeroCols.find(j) != zeroCols.end()){
                    matrix[i][j] = 0;
                }
            }
        }
    }
};
